runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/PackageUtil.java

```diff
@@ -288,45 +291,69 @@ private static void stageOnePackage(
         stagingPath != null,
         "Can't stage classpath elements because no staging location has been provided");
 
-    // Inline a copy here because the inner code returns an immutable list and we want to mutate it.
-    List<PackageAttributes> packageAttributes =
-        new LinkedList<>(computePackageAttributes(classpathElements, stagingPath, executorService));
-
-    // Compute the returned list of DataflowPackage objects here so that they are returned in the
-    // same order as on the classpath.
-    List<DataflowPackage> packages = Lists.newArrayListWithExpectedSize(packageAttributes.size());
-    for (final PackageAttributes attributes : packageAttributes) {
-      packages.add(attributes.getDataflowPackage());
-    }
-
-    // Order package attributes in descending size order so that we upload the largest files first.
-    Collections.sort(packageAttributes, new PackageUploadOrder());
     final AtomicInteger numUploaded = new AtomicInteger(0);
     final AtomicInteger numCached = new AtomicInteger(0);
+    List<ListenableFuture<DataflowPackage>> destinationPackages = new ArrayList<>();
 
-    List<ListenableFuture<?>> futures = new LinkedList<>();
-    for (final PackageAttributes attributes : packageAttributes) {
-      futures.add(executorService.submit(new Runnable() {
-        @Override
-        public void run() {
-          stageOnePackage(attributes, numUploaded, numCached, retrySleeper, createOptions);
-        }
-      }));
+    for (String classpathElement : classpathElements) {
+      DataflowPackage sourcePackage = new DataflowPackage();
```

Comment:
<!--thread_id:cc_142241284_t; commit:2eb67949d13a8f290b7d29c2898b2c1ed2dcf780; resolved:1-->
<!--section:context-quote-->
> **tgroh** wrote:
> Is the creation + value migration worth factoring into its own method?

<!--section:body-->
TBH this should be moved out to the command line parsing. It has no place here. For now I'd rather leave it as-is (this bit is a code move)